pontoon/contributors/utils.py

```diff
@@ -427,24 +427,89 @@ def get_project_locale_contribution_counts(contributions_qs):
+
+    if not year_shown:
+        start = end - relativedelta(day=1)
+        timeline_title = "Contribution activity in this month"
+    else:
+        timeline_title = "Contribution activity in the past year"
 
     if day is not None:
         start = datetime.datetime.fromtimestamp(day, tz=timezone.get_current_timezone())
         end = start + relativedelta(days=1)
         date = format_datetime(start, format="date")
         timeline_title = f"Contribution activity on { date }"
 
-    contribution_period = Q(created_at__gte=start, created_at__lte=end)
-    contributions_map = get_contributions_map(user, contribution_period)
+    contributions = {}
+
+    if year_shown:
+        # Perform query 12 times and append to contributions
+        for i in range(12):
+            month_start = end - relativedelta(months=i, day=1)
+            month_end = end - relativedelta(months=i, day=31)
+            contribution_period = Q(
+                created_at__gte=month_start, created_at__lte=month_end
+            )
+            month_start_name = month_start.strftime("%B %Y")
+            contributions_map = get_contributions_map(user, contribution_period)
+            contributions.update(
+                get_monthly_contributions(
+                    contributions_map,
+                    user,
+                    month_start,
+                    month_end,
+                    contribution_type,
+                    month_start_name,
+                )
+            )
+    else:
+        contribution_period = Q(created_at__gte=start, created_at__lte=end)
+        contributions_map = get_contributions_map(user, contribution_period)
+        contributions.update(
+            get_monthly_contributions(
+                contributions_map, user, start, end, contribution_type
+            )
+        )
 
-    # Get a list of explicit contribution types
+    # Sort object by months if displaying a year
+    if year_shown:
+        restructured = {}
+
+        for title, content in contributions.items():
+            month_name = content["month_name"]
+            type_name = content["type"]
+
+            if month_name not in restructured:
+                restructured[month_name] = {}
+
+            if title not in restructured[month_name]:
+                restructured[month_name][title] = {"type": type_name, "data": {}}
+
+            for key, value in content["data"].items():
+                if key not in restructured[month_name][title]["data"]:
+                    restructured[month_name][title]["data"][key] = value
+                else:
+                    restructured[month_name][title]["data"][key]["actions"].extend(
+                        value["actions"]
+                    )
+                    restructured[month_name][title]["data"][key]["count"] += value[
+                        "count"
+                    ]
+        return (restructured, timeline_title, year_shown)
+
+    return (contributions, timeline_title, year_shown)
```

Comment:
Does `year_shown` ever change here? If not, why is it included in the output?

Reply:
Yes, `year_shown` on line 505 is always True but `year_shown` on line 507 is always False. I included it in the output since the `timeline.html` template uses `year_shown` as an indicator on what object structure it should expect to render

Reply:
I meant that it doesn't change within the `get_contribution_timeline_data()` function. The code calling it already has access to the `show_year` value, and doesn't need to have it returned from the function.